### PR TITLE
Blog Post for TSANDS - Happy Together

### PR DESCRIPTION
**Pull request by TomSands**

### CODE REVIEW

File: _posts/2016-09-21-pair-programming-so-happy-together.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+### Why should I pair program?
+
+First, you get to learn.  As developers, we should always be intellectually curious and finding new rocks to turn over, and better ways to solve problems, and new technologies that make the world (and most importantly our jobs) easier.  You can learn from the most experienced guru as well as the most rookie newbie.  The guru knows the traps of software past, and can help you not re-invent the wheel to solve a problem.  The newbie has a un-warped curiosity, and if something isn't obvious to them when you explain it, then maybe it's not so obvious - time to think it over again.  In either case, you bring in diverse areas of experience that might know how a similar problem is solved elsewhere, and maybe you can reuse a method that is already written.  (DRY principles anyone?)
```
Comment (laurawhalin) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: change "a" to "an" in "The newbie has a un-warped curiosity"

File: _posts/2016-09-21-pair-programming-so-happy-together.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
...his code.  The best code is clearly written, easily understood, and can be picked up by anyone else on the team to extend, repair, or improve.  By pairing with someone on a part of the system where you have little knowledge, you become conversant in how that part works, and can help write parts of it yourself in the future.  The best opportunities arise, when as developers we make ourselves expendable.  That is a loaded statement, so let me unpack it a little.  If, as a developer, you are the only one who knows a key part of the code, or how to operate a key part of the system, you are less likely to be asked to branch out to other tasks and get different responsibilities.  This might seem counter-intuitive, but if you are seen as someone who write effective, bug-free, maintainable code that others can pick up and run with you are more likely to be asked to delve into a new initiative for two reasons.  First, you aren't absolutely required to work on the parts you already created, and second, you have a track record of doing things right.  In pairing, you accomplish this mission.  You share your knowledge of the code with a team mate so that you aren't the sole source of knowledge.
```
Comment (laurawhalin) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: update to "seen as someone who writes effective" (add "s" to "write")

File: _posts/2016-09-21-pair-programming-so-happy-together.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+### Should I always do it?
+
+Pairing isn't the solution to every development exercise.
+
+I find, for example, that prototyping works best alone.  The trick here is to remember that a prototype is by definition throw-away code.  You can reference it as you write a new feature, but it's basically a time for you to try and learn how things work.  It's a chance to research how the system works and read about best practices in the industry, in a safe, non-judgmental environment.
+
+You may also find that there are personalities that you can pair with and some you can't.  I personally like it when I have a very picky observer reviewing the code that I right.  I would rather fix any issues up front than have a customer find them.  Some people may be pretty blunt and abrasive in their response to things.  If you can't find an effective way of working together, then maybe it's not the right solution.
```
Comment (laurawhalin) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: "right" to "write"

File: _posts/2016-09-21-pair-programming-so-happy-together.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+### How to do it remotely?
+
+The tools available to doing pairing remotely have greatly improved in the recent years.  As with in person pairing, the two most important things are both looking at the same screen and having easy voice communication.
+
+For screen sharing, Google(TM) Hangouts is an easy solution, but sometime the screen resolution isn't enough to be effective.  Other tools have been the use of VNC and Citrix that allow for much more high fidelity screen sharing, and even shared keyboard and mouse functionality.  The key here is to not be looking at separate screens with the same code rendered locally.  It's critical that both parties are looking at the same screen so that the driver can use the mouse and cursor to discuss what she or he is doing.
```
Comment (laurawhalin) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: "sometime" to "sometimes"

File: _posts/2016-09-21-pair-programming-so-happy-together.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+categories: development
+tags: teamwork pair-programming
+comments: true
+author: Tom Sands
+author_github: TomSands
+banner_image: turtles_happy_together.jpg
+---
+
+So as a developer, which scenario seems more inviting to you.
+1.  Sitting in a quiet room with headphones on writing and debugging code?
```
Comment (laurawhalin) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: Add a line break above this ordered list so it is converted to <ol> in the HTML. It's currently not rendering.

Comment (TomSands) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: I wanted on, but added comma do put the break in the correct place.

File: _posts/2016-09-21-pair-programming-so-happy-together.md (hunk 1 above)
Comment (TomSands) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: check

File: _posts/2016-09-21-pair-programming-so-happy-together.md (hunk 2 above)
Comment (TomSands) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: check

File: _posts/2016-09-21-pair-programming-so-happy-together.md (hunk 3 above)
Comment (TomSands) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: check

File: _posts/2016-09-21-pair-programming-so-happy-together.md (hunk 4 above)
Comment (TomSands) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: check

File: _posts/2016-09-21-pair-programming-so-happy-together.md (hunk 5 above)
Comment (TomSands) [inline _posts/2016-09-21-pair-programming-so-happy-together.md]: check

**Pull request merged**